pkg/apis/workspaces/v1alpha1/projects.go

```diff
@@ -78,19 +68,37 @@ type CustomProjectSource struct {
 }
 
 type ZipProjectSource struct {
-	CommonProjectSource `json:",inline"`
+	// Zip project's source location address. Should be file path of the archive, e.g. file://$FILE_PATH
+	// +required
+	Location string `json:"location,omitempty"`
+
+	// Part of project to populate in the working directory.
+	// +optional
+	SparseCheckoutDir string `json:"sparseCheckoutDir,omitempty"`
```

Comment:
Why not keeping the `CommonProjectSource` with only `SparseCheckoutDir` inside it, instead of duplicating `SparseCheckoutDir` ?

Reply:
Good catch. I don't see any reason not to keep CommonProject Source. I'll revert it soon

Reply:
Done now.